compiler/noirc_frontend/src/monomorphization/mod.rs

```diff
@@ -1029,11 +1029,6 @@ impl<'interner> Monomorphizer<'interner> {
     }
 
     fn append_printable_type_info_inner(typ: &Type, arguments: &mut Vec<ast::Expression>) {
-        if let HirType::Array(size, _) = typ {
-            if let HirType::NotConstant = **size {
-                unreachable!("println does not support slices. Convert the slice to an array before passing it to println");
```

Comment:
Could we just add back these checks as we discussed earlier and then this looks ready to merge.

Reply:
Done, and added an extra restriction to mutable references as well.